plugins/outputs/iotdb/iotdb_test.go

```diff
@@ -0,0 +1,420 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package iotdb
+
+import (
+	"fmt"
+	"math"
+	"testing"
+	"time"
+
+	"github.com/apache/iotdb-client-go/client"
+	"github.com/stretchr/testify/require"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+var (
+	testHost     = "localhost" // The server's (ip) address that you want to connect to.
+	testPort     = "6667"      // The server's port that you want to connect to.
+	testUser     = "root"
+	testPassword = "root"
+)
+
+func TestConnectAndClose(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+	testClient := &IoTDB{
+		Host:     testHost,
+		Port:     testPort,
+		User:     testUser,
+		Password: testPassword,
+	}
+	testClient.Log = testutil.Logger{}
+
+	var err error
+	err = testClient.Connect()
+	require.NoError(t, err)
+	err = testClient.Close()
+	require.NoError(t, err)
+}
```

Comment:
Please use `testcontainers` for integration tests instead of assuming a running instance. See e.g. [`mqtt` output](https://github.com/influxdata/telegraf/blob/master/plugins/outputs/mqtt/mqtt_test.go) as an example.

Reply:
I'm not sure `testcontainers` supports IoTDB. I will have a try. In the project iotdb-client-go, there is also an E2E test as an example, which relies on docker. I will find an appropriate way. Thanks.

Reply:
Well if there is a docker container ([and there is at least one](https://hub.docker.com/r/apache/iotdb)) it is supported... :-)

Reply:
Great! I use `testcontainers` instead. Thanks a lot.